### ISSUE TITLE
Border or padding breaks absolute position height

### ISSUE THREAD

**Issue by carlobeltrame** [diegomura/react-pdf#1562]
**Describe the bug**
When setting `position: absolute` on an element, we can set the element's vertical dimensions using the `top` and `bottom` style properties. But when we also add a nonzero top or bottom border, the height of the element collapses to zero (plus the border widths).
The same thing happens when adding any vertical padding, but margin works correctly.

**To Reproduce**

[REPL](https://react-pdf.org/repl?code=3187b0760ce02e0040b204b3020b604300d80940a6007009c41805e1800a0129480f8280a061801e004446005714730a1b193160014d00731c30202005e38480220082005814c100412f2868a02708af26a804d02280a0509600d5b40772950027867901bc3e09bc7a7ef0002e18007234002308100c4e281c10801a1f2170b460006b31224e300013006168cd6090821c5cc4e4a628103c128026000600524a9b267090281a9412806616b6f6f700332812c6a19b2d31000b71d0c9a4f698007a559800214e3160a8598408184e825c9c0263b2b474a3fdc9599c0439d81011983b98003747003a2ab5d589cce041280118f0000f2934410b963860d2e929909d630000a8105c30500a0787c2418860160d2c434174e348c084b40c0e44463a6981ef6204070f71013989105988138183855c30182c460402074a05feeb2079c2690e846161f0c472298a8d12e57204982ccd0b707948d03c22440209c1c24a56eb3c1a13504b063521aa8056d49e8021889030072686ed8cd19580506165bdaeb7767ac00ea772400beb19819dc23b120794ce6d8f5884cc55bd8700e2cce7441242eadd85c3c7f1e89400373d1e8b83494184ac00188fcca7973b919888643a1b0f87a6ac68b5fa100000)

```jsx
const MinimalRepro = () => (
  <Document>
    <Page size="A4" orientation="portrait">
    <View style={{
      position: 'absolute',
      backgroundColor: 'red',
      top: '20%',
      bottom: '30%',
      left: '0',
      right: '0',
      // Bug: this border breaks the height of the view.
      //border: '1px solid black',
      // Try commenting it out or using a zero border to see how it should really look:
      //border: '0px solid black',
      // Padding has the same issue:
      //padding: '10px',
      // But margin works correctly:
      //margin: '10px'
    }} debug={true}>
    </View>
  </Page>
  </Document>
);

ReactPDF.render(<MinimalRepro />);
```

I think it's not a vanilla Yoga issue, here is the same situation recreated in the Yoga playground, working as expected: 
[Yoga playground example](https://yogalayout.com/playground/?eyJ3aWR0aCI6NTAwLCJoZWlnaHQiOjUwMCwibWluV2lkdGgiOm51bGwsIm1pbkhlaWdodCI6bnVsbCwibWF4V2lkdGgiOm51bGwsIm1heEhlaWdodCI6bnVsbCwicG9zaXRpb24iOnsidG9wIjpudWxsLCJyaWdodCI6bnVsbCwiYm90dG9tIjpudWxsLCJsZWZ0IjpudWxsfSwiY2hpbGRyZW4iOlt7IndpZHRoIjoiIiwiaGVpZ2h0IjoiIiwibWluV2lkdGgiOm51bGwsIm1pbkhlaWdodCI6bnVsbCwibWF4V2lkdGgiOm51bGwsIm1heEhlaWdodCI6bnVsbCwiYm9yZGVyIjp7InRvcCI6IjEiLCJyaWdodCI6IjEiLCJib3R0b20iOiIxIiwibGVmdCI6IjEifSwicG9zaXRpb24iOnsidG9wIjoiMjAlIiwicmlnaHQiOiIwIiwiYm90dG9tIjoiMzAlIiwibGVmdCI6IjAifSwicG9zaXRpb25UeXBlIjoxfV19)

**Expected behavior**
The border should not influence the absolute positioned element's height. Absolute positioning should determine the element's dimensions in all cases.

**Screenshots**
See REPL, try commenting out the border, or try using a border of `0px solid black`.

**Desktop (please complete the following information):**
 - OS: Ubuntu
 - Browser: Firefox
 - React-pdf version: 2.0.19

**Comment by jeetiss** [comment]
This example works fine in master.
so it was fixed here https://github.com/diegomura/react-pdf/pull/1541

<img width="688" alt="Снимок экрана 2021-10-30 в 16 59 55" src="https://user-images.githubusercontent.com/6726016/139535995-db1bb5e6-0d7b-4084-bf94-795c2b8472b1.png">

**Comment by carlobeltrame** [comment]
@jeetiss Sounds good.
I observed the problem not only when the page is the container for the absolute positioned element, but also in Views. But I see your fix made a subtle general change that will also apply to all nodes.
I have no way of testing with the current master, so I will only be able to confirm it works once the fix is released.

**Comment by jeetiss** [comment]
@carlobeltrame could you approve that bug is solved in `2.0.21`?

**Comment by carlobeltrame** [comment]
I can confirm it works with `@react-pdf/renderer@2.0.21`. Thanks a lot for the fix! :partying_face: 
The REPL page should probably be updated to use the new version.
